### ISSUE TITLE
Extend the Import from URL for zip archived imagestacks.

### ISSUE THREAD

**Issue by muratmaga** [SlicerMorph/SlicerMorph#353]
As seen from the log, as is SlicerMorph's `ImportFromURL` does automatically bring a zipfile and unzips it. But it is failing at the loading the data into the scene. 

Would it be possible to extend the functionality such that if the requested file is a zip archive, instead of trying to load the data into the scene, unzip and decide the appropriate action based on payload of the archive (i.e., for scenes, 3D models, volumes and other displayable objects -markups, segmentations, etc- load into the scene. 
For image sequences, switch to the `ImageStacks` module.

@pieper would it this be possible? With your guidance I think @smrolfe can implement this. 

Sample URL: https://app.box.com/shared/static/li017nfu3o9k77f16eqcwxw4ua025f5a.zip

```
Switch to module:  "ImportFromURL"
Requesting download li017nfu3o9k77f16eqcwxw4ua025f5a.zip from https://app.box.com/shared/static/li017nfu3o9k77f16eqcwxw4ua025f5a.zip ...
<i>Downloaded 152.2 MB (10% of 1.5 GB)...</i>
<i>Downloaded 304.4 MB (20% of 1.5 GB)...</i>
<i>Downloaded 456.6 MB (30% of 1.5 GB)...</i>
<i>Downloaded 608.8 MB (40% of 1.5 GB)...</i>
<i>Downloaded 761.0 MB (50% of 1.5 GB)...</i>
<i>Downloaded 913.1 MB (60% of 1.5 GB)...</i>
<i>Downloaded 1.0 GB (70% of 1.5 GB)...</i>
<i>Downloaded 1.2 GB (80% of 1.5 GB)...</i>
<i>Downloaded 1.3 GB (90% of 1.5 GB)...</i>
<i>Downloaded 1.5 GB (100% of 1.5 GB)...</i>
Download finished
Unzipping /Users/amaga/Library/Caches/slicer.org/Slicer/SlicerIO/li017nfu3o9k77f16eqcwxw4ua025f5a.zip into /Users/amaga/Library/Caches/slicer.org/Slicer/SlicerIO/li017nfu3o9k77f16eqcwxw4ua025f5a successful
Could not import data into the scene. Downloaded to: /Users/amaga/Library/Caches/slicer.org/Slicer/SlicerIO/li017nfu3o9k77f16eqcwxw4ua025f5a
```

**Comment by pieper** [comment]
Yes, I think that would be easy enough to add.  There's already built-in behavior to load zip files (https://github.com/Slicer/Slicer/blob/56edfa29adc1be04f3c312fe5e9db1df317e908a/Base/QTGUI/qSlicerDataDialog.cxx#L505) but probably that doesn't work for the use case like the zip you posted.  Instead you'd want to add some heuristics that inspect the zip file and pick how to load the contents (e.g. your zip has one kind of projection tiff together with the slices of the volume).

Probably it would be good to have this in the form of a scripted file reader (and maybe writer) for zip that has special cases for different conventions.  This could all be done in a python script so that it could be used consistently for local zip files in addition to downloads from URLs.

**Comment by muratmaga** [comment]
@jcfr

**Comment by muratmaga** [comment]
@amm554

**Comment by amm554** [comment]
response
